From 0a60e1020b330b26f537bd77230712b7d85d4fa5 Mon Sep 17 00:00:00 2001
From: Jianmei Zhang <jianmeiz@moqi.ai>
Date: Fri, 23 Dec 2022 14:32:10 +0800
Subject: [PATCH 020/150] Cut slowmodevectorIndex to slowvecindex with length
 less than 15, fix bug missing update last_cache_check_time

---
 src/Interpreters/Context.cpp                                  | 2 +-
 src/Storages/MergeTree/MergeTreeVectorIndexBuilderUpdater.cpp | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/Interpreters/Context.cpp b/src/Interpreters/Context.cpp
index b8936043f4..2c458d05f8 100644
--- a/src/Interpreters/Context.cpp
+++ b/src/Interpreters/Context.cpp
@@ -3150,7 +3150,7 @@ void Context::initializeBackgroundExecutorsIfNeeded()
 
     shared->slow_mode_vector_index_executor = MergeMutateBackgroundExecutor::create
     (
-        "SlowModeVectorIndex",
+        "SlowVecIndex",
         getSettingsRef().background_slow_mode_vector_pool_size,
         getSettingsRef().background_slow_mode_vector_pool_size,
         CurrentMetrics::BackgroundSlowModeVectorIndexPoolTask
diff --git a/src/Storages/MergeTree/MergeTreeVectorIndexBuilderUpdater.cpp b/src/Storages/MergeTree/MergeTreeVectorIndexBuilderUpdater.cpp
index cafee2bd36..2f61461428 100644
--- a/src/Storages/MergeTree/MergeTreeVectorIndexBuilderUpdater.cpp
+++ b/src/Storages/MergeTree/MergeTreeVectorIndexBuilderUpdater.cpp
@@ -41,6 +41,9 @@ void MergeTreeVectorIndexBuilderUpdater::removeDroppedVectorIndices(const Storag
     if (now - last_cache_check_time < RECHECK_VECTOR_INDDEX_CACHE_INTERVAL_SECONDS)
         return;
 
+    /// Update last_cache_check
+    last_cache_check_time = now;
+
     ///check existing parts to see if any cached vector index need cleaning
     std::list<std::pair<VectorIndex::CacheKey, VectorIndex::Parameters>> cached_item_list
         = VectorIndex::VectorSegmentExecutor::getAllCacheNames();
-- 
2.32.1 (Apple Git-133)

